dds/DCPS/DiscoveryBase.h

```diff
@@ -1822,6 +1878,16 @@ namespace OpenDDS {
         bit_subscriber_->lookup_datareader(DCPS::BUILT_IN_CONNECTION_RECORD_TOPIC);
       return dynamic_cast<ConnectionRecordDataReaderImpl*>(d.in());
     }
+
+    DCPS::InternalThreadBuiltinTopicDataDataReaderImpl* internal_thread_bit()
+    {
+      if (!bit_subscriber_.in())
+        return 0;
+
+      DDS::DataReader_var d =
+        bit_subscriber_->lookup_datareader(DCPS::BUILT_IN_INTERNAL_THREAD_TOPIC);
+      return dynamic_cast<InternalThreadBuiltinTopicDataDataReaderImpl*>(d.in());
```

Comment:
This looks risky, the _var will decrement the refcount when it goes out of scope, you are using the pointer after that.